Web Inspector: FolderizedTreeElement doesn't need to remember expansion states anymore since FolderTreeElement already does the job

the-chenergy opened the pull request:
#### 3cc63f452ee1f7b27315942d53543c6dd041dfb9
<pre>
Web Inspector: FolderizedTreeElement doesn&apos;t need to remember expansion states anymore since FolderTreeElement already does the job
<a href="https://rdar.apple.com/124158713">rdar://124158713</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=270595">https://bugs.webkit.org/show_bug.cgi?id=270595</a>

Reviewed by Devin Rousso.

Remove the creating and handling of expansion states from
FolderizedTreeElement.

* Source/WebInspectorUI/UserInterface/Views/FolderizedTreeElement.js:
(WI.FolderizedTreeElement):
(WI.FolderizedTreeElement.prototype.removeChildren):
- No need anymore for remembering the WI.Setting items for each created
  child FolderTreeElement.

(WI.FolderizedTreeElement.prototype._parentTreeElementForRepresentedObject):
- Passing the folderId into the created child FolderTreeElement will
  make it do the state-remembering job.

(WI.FolderizedTreeElement.prototype._folderTreeElementExpandedStateChange): Deleted.
- The child FolderTreeElement already does this job.

Canonical link: <a href="https://commits.webkit.org/276357@main">https://commits.webkit.org/276357@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/35d11618e8a1279961d59e68815cf6b80def3ee2

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/44442 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/23516 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/46891 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/47096 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/40466 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/27518 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/20910 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/36563 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/45019 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/20561 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/38260 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/17610 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/18016 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/39388 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/2492 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/40627 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/39665 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/48729 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/19417 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/15949 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/43476 "layout-tests (failure)") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/20774 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/42211 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9884 "Built successfully and passed tests") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/21101 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/20405 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

dcrousso commented on Source/WebInspectorUI/UserInterface/Views/FolderizedTreeElement.js:
```diff
@@ -298,25 +282,16 @@ WI.FolderizedTreeElement = class FolderizedTreeElement extends WI.GeneralTreeEle
 
-        folder = createFolderTreeElement.call(this, settings);
+        let folderId = `${settings.type}-${settings._folderSettingsKey}`;
+        folder = new WI.FolderTreeElement(settings.displayName, settings.representedObject, {id: folderId});
```
Style: instead of creating a `folderId` variable, we normally inline it
```js
folder = new WI.FolderTreeElement(settings.displayName, settings.representedObject, {
    id:`${settings.type}-${settings._folderSettingsKey}`,
});
```

dcrousso reviewed (approved):
thanks for the followup fix! :)

the-chenergy committed:
Web Inspector: FolderizedTreeElement doesn't need to remember expansion states anymore since FolderTreeElement already does the job
rdar://124158713
https://bugs.webkit.org/show_bug.cgi?id=270595

Reviewed by Devin Rousso.

Remove the creating and handling of expansion states from
FolderizedTreeElement.

* Source/WebInspectorUI/UserInterface/Views/FolderizedTreeElement.js:
(WI.FolderizedTreeElement):
(WI.FolderizedTreeElement.prototype.removeChildren):
- No need anymore for remembering the WI.Setting items for each created
  child FolderTreeElement.

(WI.FolderizedTreeElement.prototype._parentTreeElementForRepresentedObject):
- Passing the folderId into the created child FolderTreeElement will
  make it do the state-remembering job.

(WI.FolderizedTreeElement.prototype._folderTreeElementExpandedStateChange): Deleted.
- The child FolderTreeElement already does this job.

Canonical link: https://commits.webkit.org/276357@main
